### PR TITLE
2024.2.0 update

### PR DESCRIPTION
**Pull request by JamesRobertsonGames**
[fastparquet-feedstock](https://github.com/AnacondaRecipes/fastparquet-feedstock) 2024.2.0

**Destination channel:** defaults

### Links

- [PKG-4507](https://anaconda.atlassian.net/browse/PKG-4507) 
- [Upstream repository](https://github.com/dask/fastparquet)
- [Upstream changelog/diff](https://github.com/dask/fastparquet/releases/tag/2024.2.0)

### Explanation of changes:
- update version, enhance dependencies, update pinnings, and increase testing

[PKG-4507]: https://anaconda.atlassian.net/browse/PKG-4507?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,9 +27,11 @@ requirements:
     - pip
     - wheel
     - setuptools
+    - setuptools-scm >1.5.4
     - numpy {{ numpy }}
-    - cython 0.29
+    - cython >=0.29.23
     - pytest-runner
+    - packaging
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:34]: I don't think this is needed.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
 
 build:
   number: 0
   #skip s390x because of lack of cramjam and numba
   skip: true  # [py<38 or s390x]
-  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+  script: 
+    - {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+    # this avoids conflicts with cython built files and allows them to be imported during test phase
+    -  mv ./fastparquet ./fastparquet-src
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:19]:
```suggestion
```
Not needed.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -36,8 +41,12 @@ requirements:
     - packaging
 
 test:
+  source_files:
+    - fastparquet-src
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:45]:
```suggestion
    - fastparquet/test
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -36,8 +41,12 @@ requirements:
...
   imports:
     - fastparquet
+    - fastparquet.cencoding
+    - fastparquet.speedups
   requires:
     - pip
   commands:
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:52]:
```suggestion
  requires:
    - pip
    - pytest
  commands:
    - pytest -v fastparquet/test
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -24,9 +27,11 @@ requirements:
     - pip
     - wheel
     - setuptools
+    - setuptools-scm >1.5.4
     - numpy {{ numpy }}
-    - cython 0.29
+    - cython >=0.29.23
     - pytest-runner
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:33]:
```suggestion
```

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml [hunk 6]
```diff
@@ -36,11 +38,17 @@ requirements:
     - packaging
 
 test:
+  source_files:
+    - fastparquet/test
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:42]: The _Source Distribution_ on `PyPI` doesn't contain the `test` folder, see https://pypi.org/project/fastparquet/#files

**Comment by JamesRobertsonGames** [comment]
@JeanChristopheMorinPerso Applied your changes just to make sure my testing was correct
- Tests are excluded from the setup.py and are not here https://github.com/dask/fastparquet/blob/bb00f37f211d979cc2779dc2adfd6313b9c069d4/setup.py#L85
- Once this is fixed, there will be an issue with importing the cython dependencies unless I move the source code to another file, I believe due to a filename clash, I can't personally find another way to repair this

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml [hunk 7]
```diff
@@ -1,19 +1,20 @@
...
-  sha256: 2ff39bc20869829ea8fd85bf0336c5006d0ffcc276d0de070030cf8ffd929160
+  sha256: 81a8f60c51793eb2436b4fdbbf115ff8578a4a457a179240bc08f9d9573d57a4
 
 build:
   number: 0
   #skip s390x because of lack of cramjam and numba
   skip: true  # [py<38 or s390x]
-  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+  script: 
+    - {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:17]:
```suggestion
  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
```

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Looks good to me, thanks @JamesRobertsonGames. My suggestion is cosmetic only and doesn't have to be applied before merging.

**Pull request merged**